source/cdk/lib/media-insights-stack.ts

```diff
@@ -1478,6 +1478,7 @@ export class MediaInsightsStack extends Stack {
                 resourceType: 'CFN_STACK'
             },
         );
+        analyticsStackAssociation.cfnOptions.condition = deployAnalyticsCondition;
```

Comment:
These are good fixes. But, it will cause the CDK tests to fail. Try running `source/cdk/run-tests.sh`. We need to update the snapshots based on the change.